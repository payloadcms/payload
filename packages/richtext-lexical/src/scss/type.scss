@import 'vars';
@import 'queries';

/////////////////////////////
// HEADINGS
/////////////////////////////
@layer payload-default {
  %h1,
  %h2,
  %h3,
  %h4,
  %h5,
  %h6 {
    font-family: var(--font-body);
    font-weight: 500;
  }

  %h1 {
    margin: 0;
    font-size: base(1.6);
    line-height: base(1.8);

    @include small-break {
      letter-spacing: -0.5px;
      font-size: base(1.25);
    }
  }

  %h2 {
    margin: 0;
    font-size: base(1.3);
    line-height: base(1.6);

    @include small-break {
      font-size: base(0.85);
    }
  }

  %h3 {
    margin: 0;
    font-size: base(1);
    line-height: base(1.2);

    @include small-break {
      font-size: base(0.65);
      line-height: 1.25;
    }
  }

  %h4 {
    margin: 0;
    font-size: base(0.8);
    line-height: base(1);
    letter-spacing: -0.375px;
  }

  %h5 {
    margin: 0;
    font-size: base(0.65);
    line-height: base(0.8);
  }

  %h6 {
    margin: 0;
    font-size: base(0.6);
    line-height: base(0.8);
  }

  %small {
    margin: 0;
    font-size: 12px;
    line-height: 20px;
  }

  /////////////////////////////
  // TYPE STYLES
  /////////////////////////////

  %large-body {
    font-size: base(0.6);
    line-height: base(1);
    letter-spacing: base(0.02);

    @include mid-break {
      font-size: base(0.7);
      line-height: base(1);
    }

    @include small-break {
      font-size: base(0.55);
      line-height: base(0.75);
    }
  }

  %body {
    font-size: $baseline-body-size;
    line-height: $baseline-px;
    font-weight: normal;
    font-family: var(--font-body);
  }

  %code {
    font-size: base(0.4);
    color: var(--theme-elevation-400);

    span {
      color: var(--theme-elevation-800);
    }
  }
}
