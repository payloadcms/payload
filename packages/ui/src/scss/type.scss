@use 'vars';
@use 'queries';

/////////////////////////////
// HEADINGS
/////////////////////////////

@layer payload-default {
  %h1,
  %h2,
  %h3,
  %h4,
  %h5,
  %h6 {
    font-family: var(--font-body);
    font-weight: 500;
  }

  %h1 {
    margin: 0;
    font-size: vars.base(1.6);
    line-height: vars.base(1.8);

    @include queries.small-break {
      letter-spacing: -0.5px;
      font-size: vars.base(1.25);
    }
  }

  %h2 {
    margin: 0;
    font-size: vars.base(1.3);
    line-height: vars.base(1.6);

    @include queries.small-break {
      font-size: vars.base(0.85);
    }
  }

  %h3 {
    margin: 0;
    font-size: vars.base(1);
    line-height: vars.base(1.2);

    @include queries.small-break {
      font-size: vars.base(0.65);
      line-height: 1.25;
    }
  }

  %h4 {
    margin: 0;
    font-size: vars.base(0.8);
    line-height: vars.base(1);
    letter-spacing: -0.375px;
  }

  %h5 {
    margin: 0;
    font-size: vars.base(0.65);
    line-height: vars.base(0.8);
  }

  %h6 {
    margin: 0;
    font-size: vars.base(0.6);
    line-height: vars.base(0.8);
  }

  %small {
    margin: 0;
    font-size: 12px;
    line-height: 20px;
  }

  /////////////////////////////
  // TYPE STYLES
  /////////////////////////////

  %large-body {
    font-size: vars.base(0.6);
    line-height: vars.base(1);
    letter-spacing: vars.base(0.02);

    @include queries.mid-break {
      font-size: vars.base(0.7);
      line-height: vars.base(1);
    }

    @include queries.small-break {
      font-size: vars.base(0.55);
      line-height: vars.base(0.75);
    }
  }

  %body {
    font-size: vars.$baseline-body-size;
    line-height: vars.$baseline-px;
    font-weight: normal;
    font-family: var(--font-body);
  }

  %code {
    font-size: vars.base(0.4);
    color: var(--theme-elevation-400);

    span {
      color: var(--theme-elevation-800);
    }
  }
}
