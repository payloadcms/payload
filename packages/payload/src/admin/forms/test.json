{
  "lockedState": {
    "isLocked": true,
    "lastEditedAt": "2025-03-31T21:31:01.419Z",
    "user": {
      "createdAt": "2025-03-31T21:30:11.215Z",
      "updatedAt": "2025-03-31T21:30:11.215Z",
      "email": "dev@payloadcms.com",
      "id": "67eb09635d15fc77b1da4823",
      "loginAttempts": 0
    }
  },
  "state": {
    "title": { "value": "example post", "initialValue": "example post" },
    "renderTracker": {},
    "validateUsingEvent": {},
    "array.0.id": {
      "value": "67eb09903a4c5f0cce051442",
      "initialValue": "67eb09903a4c5f0cce051442",
      "lastRenderedPath": "array.0.id"
    },
    "array.0.richText": {},
    "array.1.id": {
      "value": "67eb09923a4c5f0cce051444",
      "initialValue": "67eb09923a4c5f0cce051444"
    },
    "array.1.richText": {},
    "array.2.id": {
      "value": "67eb0a7f3a4c5f0cce051446",
      "initialValue": "67eb0a7f3a4c5f0cce051446",
      "lastRenderedPath": "array.2.id"
    },
    "array.2.richText": {
      "lastRenderedPath": "array.2.richText",
      "customComponents": {
        "Field": [
          "$",
          "$L2",
          null,
          { "path": "array.2.richText", "children": "$L3" },
          null,
          [
            [
              "renderField",
              "webpack-internal:///(rsc)/./packages/ui/src/forms/fieldSchemasToFormState/renderField.tsx",
              156,
              128
            ],
            [
              "addFieldStatePromise",
              "webpack-internal:///(rsc)/./packages/ui/src/forms/fieldSchemasToFormState/addFieldStatePromise.ts",
              584,
              9
            ],
            [
              "eval",
              "webpack-internal:///(rsc)/./packages/ui/src/forms/fieldSchemasToFormState/iterateFields.ts",
              51,
              101
            ],
            [
              "iterateFields",
              "webpack-internal:///(rsc)/./packages/ui/src/forms/fieldSchemasToFormState/iterateFields.ts",
              15,
              12
            ],
            [
              "arrayValue.reduce.promises",
              "webpack-internal:///(rsc)/./packages/ui/src/forms/fieldSchemasToFormState/addFieldStatePromise.ts",
              125,
              107
            ],
            [
              "addFieldStatePromise",
              "webpack-internal:///(rsc)/./packages/ui/src/forms/fieldSchemasToFormState/addFieldStatePromise.ts",
              112,
              59
            ],
            [
              "eval",
              "webpack-internal:///(rsc)/./packages/ui/src/forms/fieldSchemasToFormState/iterateFields.ts",
              51,
              101
            ],
            [
              "iterateFields",
              "webpack-internal:///(rsc)/./packages/ui/src/forms/fieldSchemasToFormState/iterateFields.ts",
              15,
              12
            ],
            [
              "fieldSchemasToFormState",
              "webpack-internal:///(rsc)/./packages/ui/src/forms/fieldSchemasToFormState/index.tsx",
              36,
              79
            ],
            [
              "async buildFormState",
              "webpack-internal:///(rsc)/./packages/ui/src/utilities/buildFormState.ts",
              130,
              29
            ],
            [
              "async buildFormStateHandler",
              "webpack-internal:///(rsc)/./packages/ui/src/utilities/buildFormState.ts",
              55,
              21
            ],
            [
              "async Server.<anonymous>",
              "file:///Users/jacobfletcher/dev/payload/payload/test/dev.ts",
              1,
              2848
            ]
          ],
          0
        ]
      }
    },
    "updatedAt": {
      "value": "2025-03-31T21:30:11.221Z",
      "initialValue": "2025-03-31T21:30:11.221Z"
    },
    "createdAt": {
      "value": "2025-03-31T21:30:11.221Z",
      "initialValue": "2025-03-31T21:30:11.221Z"
    },
    "blocks": { "value": 0, "initialValue": 0, "rows": [] },
    "array": {
      "rows": [
        { "id": "67eb09903a4c5f0cce051442" },
        { "id": "67eb09923a4c5f0cce051444" },
        { "id": "67eb0a7f3a4c5f0cce051446" }
      ],
      "value": 3,
      "initialValue": 3,
      "disableFormData": true
    }
  }
}
