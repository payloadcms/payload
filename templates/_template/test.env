NODE_OPTIONS="--no-deprecation --no-experimental-strip-types"
