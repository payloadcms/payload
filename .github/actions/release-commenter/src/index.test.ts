import type * as githubModule from '@actions/github'
import type * as coreModule from '@actions/core'
import { mock } from 'node:test'

jest.mock('@actions/core')
jest.mock('@actions/github')

type Mocked<T> = {
  -readonly [P in keyof T]: T[P] extends Function ? jest.Mock<T[P]> : jest.Mocked<Partial<T[P]>>
}

const github = require('@actions/github') as jest.Mocked<Mocked<typeof githubModule>>
const core = require('@actions/core') as jest.Mocked<Mocked<typeof coreModule>>

describe('tests', () => {
  let mockOctokit: any = {}
  let currentTag: string = 'current_tag_name'

  ;(core.warning as any) = jest.fn(console.warn.bind(console))
  ;(core.error as any) = jest.fn(console.error.bind(console))

  let commentTempate: string = ''
  let labelTemplate: string | null = null
  const skipLabelTemplate: string | null = 'skip,test'
  let tagFilter: string | RegExp | null = null

  let simpleMockOctokit: any = {}

  beforeEach(() => {
    tagFilter = null
    currentTag = 'current_tag_name'
    ;(github.context as any) = {
      payload: {
        repo: {
          owner: 'owner',
          repo: 'repo',
        },
        release: {
          tag_name: currentTag,
        },
        repository: { html_url: 'http://repository' },
      },
    }

    github.getOctokit.mockReset().mockImplementationOnce(((token: string) => {
      expect(token).toBe('GITHUB_TOKEN_VALUE')
      return mockOctokit
    }) as any)
    ;(core.getInput as any).mockImplementation((key: string) => {
      if (key == 'GITHUB_TOKEN') {
        return 'GITHUB_TOKEN_VALUE'
      }
      if (key == 'comment-template') {
        return commentTempate
      }
      if (key == 'label-template') {
        return labelTemplate
      }
      if (key == 'skip-label') {
        return skipLabelTemplate
      }
      if (key == 'tag-filter') {
        return tagFilter
      }
      fail(`Unexpected input key ${key}`)
    })

    commentTempate =
      'Included in release {release_link}. Replacements: {release_name}, {release_tag}.'
    labelTemplate = null
    simpleMockOctokit = {
      rest: {
        issues: {
          get: jest.fn(() => Promise.resolve({ data: { locked: false } })),
          createComment: jest.fn(() => Promise.resolve()),
          addLabels: jest.fn(() => Promise.resolve()),
        },
        repos: {
          listReleases: jest.fn(() =>
            Promise.resolve({
              data: [
                {
                  name: 'Release Name',
                  tag_name: 'current_tag_name',
                  html_url: 'http://current_release',
                },
                {
                  tag_name: 'prior_tag_name',
                  html_url: 'http://prior_release',
                },
              ],
            }),
          ),
          compareCommits: jest.fn(() =>
            Promise.resolve({
              data: { commits: [{ sha: 'SHA1' }] },
            }),
          ),
        },
      },
      graphql: jest.fn(() =>
        Promise.resolve({
          resource: {
            messageHeadlineHTML: '',
            messageBodyHTML:
              '<span class="issue-keyword tooltipped tooltipped-se" aria-label="This commit closes issue #123.">Closes</span> <p><span class="issue-keyword tooltipped tooltipped-se" aria-label="This pull request closes issue #7.">Closes</span>',
            associatedPullRequests: {
              pageInfo: { hasNextPage: false },
              edges: [],
            },
          },
        }),
      ),
    }
  })

  afterEach(() => {
    expect(core.error).not.toHaveBeenCalled()
    expect(core.warning).not.toHaveBeenCalled()
    expect(core.setFailed).not.toHaveBeenCalled()
  })

  test('main test', async () => {
    mockOctokit = {
      ...simpleMockOctokit,
      rest: {
        issues: {
          get: jest.fn(() => Promise.resolve({ data: { locked: false } })),
          createComment: jest.fn(() => Promise.resolve()),
          addLabels: jest.fn(() => Promise.resolve()),
        },
        repos: {
          listReleases: jest.fn(() =>
            Promise.resolve({
              data: [
                {
                  tag_name: 'current_tag_name',
                  html_url: 'http://current_release',
                },
                {
                  tag_name: 'prior_tag_name',
                  html_url: 'http://prior_release',
                },
              ],
            }),
          ),
          compareCommits: jest.fn(() =>
            Promise.resolve({
              data: { commits: [{ sha: 'SHA1' }, { sha: 'SHA2' }] },
            }),
          ),
        },
      },
      graphql: jest.fn(() =>
        Promise.resolve({
          resource: {
            messageHeadlineHTML:
              '<span class="issue-keyword tooltipped tooltipped-se" aria-label="This commit closes issue #3.">Closes</span> <a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="718013420" data-permission-text="Title is private" data-url="https://github.com/apexskier/github-release-commenter/issues/1" data-hovercard-type="issue" data-hovercard-url="/apexskier/github-release-commenter/issues/1/hovercard" href="https://github.com/apexskier/github-release-commenter/issues/1">#1</a>',
            messageBodyHTML:
              '<span class="issue-keyword tooltipped tooltipped-se" aria-label="This commit closes issue #123.">Closes</span> <p><span class="issue-keyword tooltipped tooltipped-se" aria-label="This pull request closes issue #7.">Closes</span>',
            associatedPullRequests: {
              pageInfo: { hasNextPage: false },
              edges: [
                {
                  node: {
                    bodyHTML:
                      '<span class="issue-keyword tooltipped tooltipped-se" aria-label="This commit closes issue #4.">Closes</span> <span class="issue-keyword tooltipped tooltipped-se" aria-label="This commit closes issue #5.">Closes</span>',
                    number: 9,
                    labels: {
                      pageInfo: { hasNextPage: false },
                      nodes: [{ name: 'label1' }, { name: 'label2' }],
                    },
                    timelineItems: {
                      pageInfo: { hasNextPage: false },
                      nodes: [
                        {
                          isCrossRepository: true,
                          __typename: 'ConnectedEvent',
                          subject: { number: 1 },
                        },
                        {
                          isCrossRepository: false,
                          __typename: 'ConnectedEvent',
                          subject: { number: 2 },
                        },
                        {
                          isCrossRepository: false,
                          __typename: 'DisconnectedEvent',
                          subject: { number: 2 },
                        },
                        {
                          isCrossRepository: false,
                          __typename: 'ConnectedEvent',
                          subject: { number: 2 },
                        },
                      ],
                    },
                  },
                },
                {
                  node: {
                    bodyHTML: '',
                    number: 42,
                    labels: {
                      pageInfo: { hasNextPage: false },
                      nodes: [{ name: 'label1' }, { name: 'skip' }],
                    },
                    timelineItems: {
                      pageInfo: { hasNextPage: false },
                      nodes: [
                        {
                          isCrossRepository: true,
                          __typename: 'ConnectedEvent',
                          subject: { number: 82 },
                        },
                      ],
                    },
                  },
                },
              ],
            },
          },
        }),
      ),
    }

    jest.isolateModules(() => {
      require('./index')
    })

    await new Promise<void>(setImmediate)

    expect(mockOctokit).toMatchSnapshot()
    expect(mockOctokit.rest.issues.createComment).toHaveBeenCalledTimes(3)
  })

  describe('can filter tags', () => {
    const v3prev = 'v3.0.1'
    const v3current = 'v3.0.2'
    const v2prev = 'v2.0.1'
    const v2current = 'v2.0.2'

    const listReleasesData = [
      {
        name: 'Current Release Name',
        tag_name: v3current,
        html_url: 'http://v3.0.2',
      },
      {
        name: 'Prev Release Name',
        tag_name: v3prev,
        html_url: 'http://v3.0.1',
      },
      {
        name: 'v2 Current Release Name',
        tag_name: v2current,
        html_url: 'http://v2.0.2',
      },
      {
        name: 'v2 Prev Release Name',
        tag_name: v2prev,
        html_url: 'http://v2.0.1',
      },
    ]

    it.each`
      description    | prevTag   | currentTag   | filter
      ${'no filter'} | ${v3prev} | ${v3current} | ${null}
      ${'v3'}        | ${v3prev} | ${v3current} | ${'v\\d'}
      ${'v2'}        | ${v2prev} | ${v2current} | ${'v\\d'}
    `('should filter tags with $description', async ({ prevTag, currentTag, filter }) => {
      // @ts-ignore
      github.context.payload.release.tag_name = currentTag

      tagFilter = filter

      mockOctokit = {
        ...simpleMockOctokit,
        rest: {
          issues: {
            get: jest.fn(() => Promise.resolve({ data: { locked: false } })),
            createComment: jest.fn(() => Promise.resolve()),
            addLabels: jest.fn(() => Promise.resolve()),
          },
          repos: {
            listReleases: jest.fn(() =>
              Promise.resolve({
                data: listReleasesData,
              }),
            ),
            compareCommits: jest.fn(() =>
              Promise.resolve({
                data: { commits: [{ sha: 'SHA1' }] },
              }),
            ),
          },
        },
        graphql: jest.fn(() =>
          Promise.resolve({
            resource: {
              messageHeadlineHTML: '',
              messageBodyHTML:
                '<span class="issue-keyword tooltipped tooltipped-se" aria-label="This commit closes issue #123.">Closes</span> <p><span class="issue-keyword tooltipped tooltipped-se" aria-label="This pull request closes issue #7.">Closes</span>',
              associatedPullRequests: {
                pageInfo: { hasNextPage: false },
                edges: [],
              },
            },
          }),
        ),
      }

      jest.isolateModules(() => {
        require('./index')
      })

      await new Promise<void>(resolve => setImmediate(() => resolve()))

      expect(github.getOctokit).toHaveBeenCalled()
      expect(mockOctokit.rest.repos.compareCommits.mock.calls).toEqual([
        [{ base: prevTag, head: currentTag }],
      ])
    })
  })

  describe('feature tests', () => {
    beforeEach(() => {
      mockOctokit = simpleMockOctokit
    })

    it('can disable comments', async () => {
      commentTempate = ''

      jest.isolateModules(() => {
        require('./index')
      })

      await new Promise<void>(resolve => setImmediate(() => resolve()))

      expect(github.getOctokit).toHaveBeenCalled()
      expect(mockOctokit.rest.issues.createComment).not.toHaveBeenCalled()
    })

    it('should unlock and comment', async () => {
      mockOctokit = {
        ...simpleMockOctokit,
        rest: {
          ...simpleMockOctokit.rest,
          issues: {
            // Return locked for both issues to be commented on
            get: jest.fn(() => Promise.resolve({ data: { locked: true } })),
            lock: jest.fn(() => Promise.resolve()),
            unlock: jest.fn(() => Promise.resolve()),
            createComment: jest.fn(() => Promise.resolve()),
          },
        },
        graphql: jest.fn(() =>
          Promise.resolve({
            resource: {
              messageHeadlineHTML: '',
              messageBodyHTML:
                '<span class="issue-keyword tooltipped tooltipped-se" aria-label="This commit closes issue #123.">Closes</span> <p><span class="issue-keyword tooltipped tooltipped-se" aria-label="This pull request closes issue #7.">Closes</span>',
              associatedPullRequests: {
                pageInfo: { hasNextPage: false },
                edges: [],
              },
            },
          }),
        ),
      }

      jest.isolateModules(() => {
        require('./index')
      })

      await new Promise<void>(resolve => setImmediate(() => resolve()))

      expect(github.getOctokit).toHaveBeenCalled()

      // Should call once for both linked issues
      expect(mockOctokit.rest.issues.unlock).toHaveBeenCalledTimes(2)
      expect(mockOctokit.rest.issues.createComment).toHaveBeenCalledTimes(2)
      expect(mockOctokit.rest.issues.lock).toHaveBeenCalledTimes(2)
    })

    it.skip('can apply labels', async () => {
      labelTemplate = ':dart: landed,release-{release_tag},{release_name}'

      jest.isolateModules(() => {
        require('./index')
      })

      await new Promise<void>(resolve => setImmediate(() => resolve()))

      expect(github.getOctokit).toHaveBeenCalled()
      expect(mockOctokit.rest.issues.addLabels.mock.calls).toMatchSnapshot()
    })
  })
})
