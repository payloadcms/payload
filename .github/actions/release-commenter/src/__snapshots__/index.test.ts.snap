// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests feature tests can apply labels 1`] = `
[
  [
    {
      "issue_number": 123,
      "labels": [
        ":dart: landed",
        "release-current_tag_name",
        "Release Name",
      ],
    },
  ],
  [
    {
      "issue_number": 7,
      "labels": [
        ":dart: landed",
        "release-current_tag_name",
        "Release Name",
      ],
    },
  ],
]
`;

exports[`tests main test 1`] = `
{
  "graphql": [MockFunction] {
    "calls": [
      [
        "
            {
              resource(url: "http://repository/commit/SHA1") {
                ... on Commit {
                  messageHeadlineHTML
                  messageBodyHTML
                  associatedPullRequests(first: 10) {
                    pageInfo {
                      hasNextPage
                    }
                    edges {
                      node {
                        bodyHTML
                        number
                        state
                        labels(first: 10) {
                          pageInfo {
                            hasNextPage
                          }
                          nodes {
                            name
                          }
                        }
                        timelineItems(itemTypes: [CONNECTED_EVENT, DISCONNECTED_EVENT], first: 100) {
                          pageInfo {
                            hasNextPage
                          }
                          nodes {
                            ... on ConnectedEvent {
                              __typename
                              isCrossRepository
                              subject {
                                ... on Issue {
                                  number
                                }
                              }
                            }
                            ... on DisconnectedEvent {
                              __typename
                              isCrossRepository
                              subject {
                                ... on Issue {
                                  number
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          ",
      ],
      [
        "
            {
              resource(url: "http://repository/commit/SHA2") {
                ... on Commit {
                  messageHeadlineHTML
                  messageBodyHTML
                  associatedPullRequests(first: 10) {
                    pageInfo {
                      hasNextPage
                    }
                    edges {
                      node {
                        bodyHTML
                        number
                        state
                        labels(first: 10) {
                          pageInfo {
                            hasNextPage
                          }
                          nodes {
                            name
                          }
                        }
                        timelineItems(itemTypes: [CONNECTED_EVENT, DISCONNECTED_EVENT], first: 100) {
                          pageInfo {
                            hasNextPage
                          }
                          nodes {
                            ... on ConnectedEvent {
                              __typename
                              isCrossRepository
                              subject {
                                ... on Issue {
                                  number
                                }
                              }
                            }
                            ... on DisconnectedEvent {
                              __typename
                              isCrossRepository
                              subject {
                                ... on Issue {
                                  number
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          ",
      ],
    ],
    "results": [
      {
        "type": "return",
        "value": Promise {},
      },
      {
        "type": "return",
        "value": Promise {},
      },
    ],
  },
  "rest": {
    "issues": {
      "addLabels": [MockFunction],
      "createComment": [MockFunction] {
        "calls": [
          [
            {
              "body": "Included in release [current_tag_name](http://current_release). Replacements: current_tag_name, current_tag_name.",
              "issue_number": 3,
            },
          ],
          [
            {
              "body": "Included in release [current_tag_name](http://current_release). Replacements: current_tag_name, current_tag_name.",
              "issue_number": 123,
            },
          ],
          [
            {
              "body": "Included in release [current_tag_name](http://current_release). Replacements: current_tag_name, current_tag_name.",
              "issue_number": 7,
            },
          ],
        ],
        "results": [
          {
            "type": "return",
            "value": Promise {},
          },
          {
            "type": "return",
            "value": Promise {},
          },
          {
            "type": "return",
            "value": Promise {},
          },
        ],
      },
      "get": [MockFunction] {
        "calls": [
          [
            {
              "issue_number": 3,
            },
          ],
          [
            {
              "issue_number": 123,
            },
          ],
          [
            {
              "issue_number": 7,
            },
          ],
        ],
        "results": [
          {
            "type": "return",
            "value": Promise {},
          },
          {
            "type": "return",
            "value": Promise {},
          },
          {
            "type": "return",
            "value": Promise {},
          },
        ],
      },
    },
    "repos": {
      "compareCommits": [MockFunction] {
        "calls": [
          [
            {
              "base": "prior_tag_name",
              "head": "current_tag_name",
            },
          ],
        ],
        "results": [
          {
            "type": "return",
            "value": Promise {},
          },
        ],
      },
      "listReleases": [MockFunction] {
        "calls": [
          [
            {
              "per_page": 100,
            },
          ],
        ],
        "results": [
          {
            "type": "return",
            "value": Promise {},
          },
        ],
      },
    },
  },
}
`;
